File: 02_imperative/c/linear_search.c
#include <stdio.h>

int main()
{
  int my_list[9]={1,2,32,8,17,19,42,13,0};
  int my_key = 3;
  int i;
  for (i = 0; i < sizeof(my_list); i++)
  {
    if (my_list[i] == my_key)
    {
      printf("%d found at location %d", my_key, i+1);
      return 0;
    }
  }
  printf("%d not found in the array\n", my_key);
  return 0;
}

File: 02_imperative/python/linear_search.py
#!/usr/bin/env python

def linear_search(alist, key):
    i = 0
    found = False

    while i < len(alist) and not found:
        if alist[i] == key:
            found = True
        else:
            i = i+1

    return found

my_list = [1, 2, 32, 8, 17, 19, 42, 13, 0]
search_for = 3
print("Searching for ", 3, " in ", my_list)
print("Does it exist?", linear_search(my_list, search_for))
